data/model/user.py

```diff
@@ -789,34 +794,69 @@ def validate_reset_code(token):
 
 
 def find_user_by_email(email):
+    # Make sure we didn't get any unicode for the username.
+    try:
+        if email and isinstance(email, str):
+            email.encode("ascii")
+    except UnicodeEncodeError:
+        return None
```

Comment:
Why do we need to do this conversion? We shouldn't be allowing non ascii characters in the first place.
You end up dealing with a n instance of bytes as opposed to a string after this encoding.

Reply:
IIRC, there was a unit test testing a user with Swedish characters in their username and email which was failing without this.

Reply:
Found it, it was the `test_signin_unicode` in `test_api_usage.py` that tried to login with Unicode characters. In endpoints/api/user.py I am using `model.user.find_user_by_email()` there to attempt to detect the performing user. This tests fails when the registry unit tests are run on a MySQL database. The call to `find_user_by_email(()` wasn't there before but `get_nonrobot_user()` in the user model has the same check, most likely for the same reason.